openid-4-verifiable-presentations-1_0.md

```diff
@@ -261,9 +261,13 @@ This specification defines the following new parameters:
 : OPTIONAL. A string identifying the scheme of the value in the `client_id` Authorization Request parameter (Client Identifier scheme). The `client_id_scheme` parameter namespaces the respective Client Identifier. If an Authorization Request uses the `client_id_scheme` parameter, the Wallet MUST interpret the Client Identifier of the Verifier in the context of the Client Identifier scheme. If the parameter is not present, the Wallet MUST behave as specified in [@!RFC6749]. See (#client_metadata_management) for the values defined by this specification. If the same Client Identifier is used with different Client Identifier schemes, those occurrences MUST be treated as different Verifiers. Note that the Verifier needs to determine which Client Identifier schemes the Wallet supports prior to sending the Authorization Request in order to choose a supported scheme.
 
 `client_metadata`:
-: OPTIONAL. A JSON object containing the Verifier metadata values. It MUST be UTF-8 encoded.
+: OPTIONAL. A JSON object containing the Verifier metadata values. It MUST be UTF-8 encoded. It may contain the following:
```

Comment:
"It MAY contain the following:" is somewhat ambiguous.

Is it the intent to say
"It MAY contain the following, other parameters MUST NOT be used."
or
"It MAY contain the following, other parameters MAY be used."
Or something else?

Reply:
Thanks for the feedback Martijn!

It's actually the 'something else' option, other values MUST NOT be used (because, for example, a self-asserted `client_name` passed in the authorization request just opens up phishing attacks without really achieving any useful) - but we do want to leave it open for other specifications to define new values that can be used.

I've pushed new text, please let me know what you think of it.

Reply:
Is the following an accurate interpretation of that?

OpenID4VP defines that the client_metadata parameter contains metadata with values defined in:
- Section 4.3 of the OpenID4VP specification
- Section 2.1 of the OpenID Connect dynamic Client Registration specification 1.0 (should this be section 2.0?)
- RFC 7591 (should we explicitly call out section 2 here?)

The intent is to say that out of the values that are defined in these sections, only the 5 parameters defined in the PR are allowed to be used and others are explicitly not allowed. But any value that's not on the list of the referenced section may also be used, but their use is out of scope of the OpenID4VP specification?

And the following is somewhat pedantic, but if there is a PR it's good to clear it up.

- The OpenID4VP specification references section mentions "OpenID Connect Dynamic Client Registration 1.0 incorporating errata set 1", 8 November 2014,", but the link links to "https://openid.net/specs/openid-connect-registration-1_0.html" which incorporates errata set 2.
- Both errata 1 and errata 2 have a line saying that additional client metadata values are allowed. Specific specs are referenced between errata 1 and errata 2. Are the parameters in these specifications ([OpenID Connect Session Management 1.0 and [OpenID Connect RP-Initiated Logout 1.0]) also not allowed because they are referenced in the client registration specification?